src/common/libflux/message.c

```diff
@@ -416,18 +424,26 @@ static int msg_append_route (flux_msg_t *msg,
     return 0;
 }
 
-static int zmsg_to_msg (flux_msg_t *msg, zmsg_t *zmsg)
+static int iovec_to_msg (flux_msg_t *msg,
+                         struct msg_iovec *iov,
+                         int iovcnt)
 {
-    uint8_t *proto_data;
+    unsigned int index = 0;
+    const uint8_t *proto_data;
     size_t proto_size;
-    zframe_t *zf;
 
-    if (!(zf = zmsg_last (zmsg))) {
+    assert (msg);
+    assert (iov);
+    assert (iovcnt >= 0);
```

Comment:
Should this assertion on iovcnt be `>` not `>=` ?

Reply:
yeah, this assert should be removed.  The `if (!iovcnt)` you mention above should just capture the 0 case.